lib/puppet/util/logging.rb

```diff
@@ -67,7 +67,7 @@ def log_exception(exception, message = :default, options = {})
           :node => exception.node
         }.merge(log_metadata))
     else
-      err(format_exception(exception, message, trace))
+      send_log(options[:level] || :err, format_exception(exception, message, trace))
```

Comment:
Ideally nothing will ever raise an exception during this, so logging at `info` level instead of letting it bubble up is an acknowledgement that we're doing new/weird things, and that not all code will be compatible.

By logging the full exception and allowing `--trace` to work correctly, we provide the information that module authors need to make their code play nice with this functionality, if so desired.